okonomiyaki/file_formats/_package_info.py

```diff
@@ -69,7 +74,10 @@ def from_egg(cls, path_or_file):
                 raise OkonomiyakiError(msg)
         else:
             data = path_or_file.read(_PKG_INFO_LOCATION)
-            data = data.decode(PKG_INFO_ENCODING)
+            with _keep_position(path_or_file.fp):
+                sha256 = compute_sha256(_keep_position(path_or_file.fp))
```

Comment:
On second thought ... is this even tested? you're computing the sha256 on a generator argument, not a file.